packages/nextjs/app/api/og/route.tsx

```diff
@@ -0,0 +1,269 @@
+/* eslint-disable @next/next/no-img-element */
+import { ImageResponse } from "next/og";
+import { blo } from "blo";
+import { promises as fs } from "fs";
+import { Address, createPublicClient, getAddress, http, isAddress } from "viem";
+import { mainnet } from "viem/chains";
+import { normalize } from "viem/ens";
+import {
+  UserChallenges,
+  getLatestSubmissionPerChallengeByUser,
+} from "~~/services/database/repositories/userChallenges";
+import { getAlchemyHttpUrl } from "~~/utils/scaffold-eth";
+
+const alchemyHttpUrl = getAlchemyHttpUrl(mainnet.id);
+const rpcUrl = alchemyHttpUrl;
+const publicClient = createPublicClient({
+  chain: mainnet,
+  transport: http(rpcUrl),
+});
+
+export async function GET(request: Request) {
+  try {
+    const { searchParams } = new URL(request.url);
+    const address = searchParams.get("address") || "0x0000000000000000000000000000000000000000";
+    const theme = searchParams.get("theme") || "default";
+
+    if (!isAddress(address)) {
+      return new Response("Invalid Ethereum address format", { status: 400 });
+    }
+
+    const checksumAddress = getAddress(address);
+
+    let ensName = null;
+    try {
+      ensName = await publicClient.getEnsName({ address: checksumAddress as Address });
+    } catch (error) {
+      console.error("Error resolving ENS name:", error);
+    }
+
+    let challenges: UserChallenges = [];
+    try {
+      challenges = await getLatestSubmissionPerChallengeByUser(address);
+    } catch (error) {
+      console.error("Error fetching builder challenges", error);
+    }
```

Comment:
We need to filter out the challenges that are not ACCEPTED. Check `enum ReviewAction`

Reply:
Ah thanks, fixed via https://github.com/BuidlGuidl/SpeedRunEthereum-v2/pull/120/commits/645ebd5768a70932d08dc034588557781c2c8e1f